### ISSUE TITLE
Add Printer Support

### ISSUE THREAD

**Issue by michael-manley** [sarah-walker-pcem/pcem#91]
This is more of a nice to have, but add at least Text Mode and ESC/P support first.

I did just commit the "Very Minimal" and made an addition to extend LPT devices. Added to experimental though for now, don't want it in
releases as its unusable at the moment.

**Comment by eluancm** [comment]
Have you checked https://pcem-emulator.co.uk/phpBB3/viewtopic.php?f=4&t=3641?

I last rebased it to v17. It's very good apart from some hacky font stuff. It implements the ESC/P dialect spoken by the LX-810.

[lx810_diff_rebased_to_v17.patch.txt](https://github.com/sarah-walker-pcem/pcem/files/7879828/lx810_diff_rebased_to_v17.patch.txt)

**Comment by michael-manley** [comment]
@eluancm I saw it, and adapted it to work on dev.

It is in commit 5b23875cf5a8e0d7fa7e3070f49d1a13e4adf0fb

It isn't in the action builds, as its flagged experimental, but you can check it out with a build with the flags `-DUSE_EXPERIMENTAL=ON -DUSE_EXPERIMENTAL_PRINTER=ON` enabled

**Comment by michael-manley** [comment]
I am leaving open for the following changes before I want to make this no longer an experimental idea

- Get rid of the Freetype dependency. It is good for a proof of concept, but i rather it use the font rom from a actual LX-810
- I might get rid of the direct to real printer, if it prints as a PDF to be honest what harm is 1 extra step to print to a real printer, im not 100% on this one yet.

**Comment by eluancm** [comment]
Completely agree on the Freetype part, I started researching this but stopped a long time ago.

About the real printer redirection, the original code from DOSBox is Windows-only anyway, so I couldn't test. I was planning on doing it for Linux to make PCem more usable in legacy situations (DOSBox and some virtual LPT drivers are used a lot). This would only add runtime dependencies, but this may not be PCem's aim. :-)

I also see that you didn't merge the additions to the menu with the printer's physical buttons. They are essential for very old programs and console printing because they don't know how to configure the printer and the buffer is only flushed to the host's filesystem after a page is complete (these are line printers). My original idea was to move these front-panel buttons to a new window with a preview of the current (un-flushed) page's buffer (and maybe a view of the flushed pages too).

**Comment by michael-manley** [comment]
@eluancm Yeah I know the buttons are needed, but I have yet to create an extension to add to the wxWidgets Menu in plugins so that's why they are not there yet.

There is a reason its in experimental (and due to major API changes, don't even build right now with experimental on)

**Comment by marcavalos01-prog** [comment]
Hi, This is Marc. I Tried to Make a Build an Experimental Printer for PCem. I Tried Msys2, Git Bash, CMake, WxWidgets. Nothing Works! Can Someone Help Me How to Do This? Or Maybe Someone will do this Build for Me. And, There's Only One Last Thing that I Have to Say... There's Nothing I Can Do. (Just Like Bill the Lab Guy from Action League Now! That He Always Said that.) Please Help Me When You Get a Chance. I Appreciate That.
